File: campus-navigate-app/expo-app/src/components/map/actor/viewHeadingModel.ts
export function closestViewHeadingTarget(
  currentHeading: number,
  nextHeading: number,
) {
  const delta = ((((nextHeading - currentHeading) % 360) + 540) % 360) - 180;
  return currentHeading + delta;
}

export function resolveUserFacingHeadingDegrees(input: {
  lastObservedHeadingDegrees: number;
  observedHeadingDegrees?: number | null;
}) {
  return input.observedHeadingDegrees ?? input.lastObservedHeadingDegrees;
}

File: tarumt-nav-app/flutter_app/lib/ui/map/models/view_heading_model.dart
double closestViewHeadingTarget(double currentHeading, double nextHeading) {
  final wrappedDifference = (nextHeading - currentHeading) % 360;
  final delta = ((wrappedDifference + 540) % 360) - 180;
  return currentHeading + delta;
}

double resolveUserFacingHeadingDegrees({
  required double lastObservedHeadingDegrees,
  required double? observedHeadingDegrees,
}) {
  return observedHeadingDegrees ?? lastObservedHeadingDegrees;
}
